lib/utils.js

```diff
@@ -27,6 +27,26 @@ const utils = (module.exports = {
       return a === b;
     }),
 
+  duplicateArrayLikeUsingType(obj, type) {
```

Comment:
@papandreou I've been thinking about this on and off for a few days now.. part of me think running through getKeys() makes sense, but I'm also quite hesitant - IIRC at the point where the arrays are compared they it never previously ran through getKeys and something vague keeps nagging me I did it this way on purpose.

Reply:
Since `duplicateArrayLikeUsingType` is being used in `<array-like> to satisfy <array-like>`, I think it might break if you have a subtype of `array-like` that has a custom `getKeys` and non-enumerable properties that are included by the `getKeys`. In that case I think you'll get an incomplete diff rendered if `getKeys` is not consulted.

Something like (untested):

```js
expect.addType({
  name: 'foo',
  base: 'array-like',
  identify(obj) {
   return obj && obj._isFoo;
  },
  getKeys(obj) {
    return [...Object.keys(obj), 'bar'];
  }
});

const foo1 = ['hey', 'there'];
const foo2 = ['hey', 'there'];
Object.defineProperty(foo1, 'bar', {
  value: 123
});
Object.defineProperty(foo2, 'bar', {
  value: 456
});

expect(foo, 'to satisfy', bar); // will fail but the diff won't mention the bar attributes
```

Reply:
@papandreou in that case this might actually be a pre-existing niggle - note that the unprocessed arrays were passed into arrayChanges previously:
https://github.com/unexpectedjs/unexpected/pull/423/files#diff-0a56a2b979f6080486ec1909508cda1aR1539

I had thought there was some other mechanism that factored getKeys into the mix but perhaps not. Will look into this and try to reach a conclusion one way or the other.